genprotocol/sserver-avdl/sserver.avdl

```diff
@@ -12,9 +12,15 @@ protocol searchServer {
     long size;
   }
 
+  record Trapdoor {
+    array<bytes> codeword;
+    int keyGen;
+  }
+
   void writeIndex(FolderID tlfID, bytes secureIndex, DocumentID docID);
   void renameIndex(FolderID tlfID, DocumentID orig, DocumentID curr);
   void deleteIndex(FolderID tlfID, DocumentID docID);
-  array<DocumentID> searchWord(FolderID tlfID, array<bytes> trapdoors);
+  array<int> getKeyGens(FolderID tlfID);
+  array<DocumentID> searchWord(FolderID tlfID, map<Trapdoor> trapdoors);
```

Comment:
Why does `trapdoors` need to be a map instead of an array, and/or why does the key gen also need to be in the `Trapdoor` record?  Seems like it's duplicating info.